.gitignore

```diff
@@ -29,3 +29,7 @@ docker-compose/docker-compose-parallel.yml
 
 # pgindent typedef lists
 *.list
+
+# Node
+**/node_modules/
+**/package-lock.json
```

Comment:
Can't we do this without Node/js? I really don't like having to install another scripting env just so I can _(checks notes)_ lint some files.

If we must work with nodejs, then I have one other question: Why gitignore the package-lock.json? Isn't it good practice to lock in dependencies that are executed in CI, so as to limit the susceptability to supply chain attacks in CI?

Reply:
We're already doing it with Node.js, this PR unifies the way we do it. If we can find alternatives for `@sourcemeta/jsonschema` and `@redocly/cli` we can replace them (in another PR).

Good point regarding `package-lock.json`, added it back!